src/Compilers/CSharp/Portable/Syntax/SyntaxExtensions.cs

```diff
@@ -70,15 +70,18 @@ public static SyntaxToken NormalizeWhitespace(this SyntaxToken token, string ind
         /// </summary>
         internal static TypeSyntax Type(this DeclarationExpressionSyntax self)
         {
-            return self.Declaration.Type;
+            var component = (TypedVariableComponentSyntax)self.Declaration;
```

Comment:
will this cast always succeed?  If not, i'd prefer a check and exception thrown with a good message (like "this can only be called on a blah blah blah").